core/src/main/java/org/apache/lucene/queryparser/classic/QueryParserSettings.java

```diff
@@ -19,64 +19,72 @@
 
 package org.apache.lucene.queryparser.classic;
 
-import com.carrotsearch.hppc.ObjectFloatHashMap;
 import org.apache.lucene.analysis.Analyzer;
-import org.apache.lucene.search.FuzzyQuery;
 import org.apache.lucene.search.MultiTermQuery;
-import org.apache.lucene.util.automaton.Operations;
 import org.elasticsearch.common.unit.Fuzziness;
 import org.joda.time.DateTimeZone;
 
-import java.util.List;
 import java.util.Locale;
+import java.util.Map;
 
 /**
- *
+ * Encapsulates settings that affect query_string parsing via {@link MapperQueryParser}
  */
 public class QueryParserSettings {
 
-    public static final boolean DEFAULT_ALLOW_LEADING_WILDCARD = true;
-    public static final boolean DEFAULT_ANALYZE_WILDCARD = false;
-    public static final float DEFAULT_BOOST = 1.f;
+    private final String queryString;
 
-    private String queryString;
     private String defaultField;
-    private float boost = DEFAULT_BOOST;
-    private MapperQueryParser.Operator defaultOperator = QueryParser.Operator.OR;
-    private boolean autoGeneratePhraseQueries = false;
-    private boolean allowLeadingWildcard = DEFAULT_ALLOW_LEADING_WILDCARD;
-    private boolean lowercaseExpandedTerms = true;
-    private boolean enablePositionIncrements = true;
-    private int phraseSlop = 0;
-    private Fuzziness fuzziness = Fuzziness.AUTO;
-    private int fuzzyPrefixLength = FuzzyQuery.defaultPrefixLength;
-    private int fuzzyMaxExpansions = FuzzyQuery.defaultMaxExpansions;
-    private int maxDeterminizedStates = Operations.DEFAULT_MAX_DETERMINIZED_STATES;
-    private MultiTermQuery.RewriteMethod fuzzyRewriteMethod = null;
-    private boolean analyzeWildcard = DEFAULT_ANALYZE_WILDCARD;
-    private boolean escape = false;
-    private Analyzer defaultAnalyzer = null;
-    private Analyzer defaultQuoteAnalyzer = null;
-    private Analyzer forcedAnalyzer = null;
-    private Analyzer forcedQuoteAnalyzer = null;
-    private String quoteFieldSuffix = null;
-    private MultiTermQuery.RewriteMethod rewriteMethod = MultiTermQuery.CONSTANT_SCORE_FILTER_REWRITE;
-    private String minimumShouldMatch;
-    private boolean lenient;
+
+    private Map<String, Float> fieldsAndWeights;
```

Comment:
I think the builder should hold all the default values, this one is just a convenient object used to provide all of the arguments to the `MapperQueryParser`. Everything should be set and passed over to this object as part of the `toQuery`. Would be nice if everything was final but then we would big constructor with lots of arguments, shall I do that instead?